pkg/sdk/schemas_test.go

```diff
@@ -0,0 +1,285 @@
+package sdk
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSchemasCreate(t *testing.T) {
+	t.Run("clone", func(t *testing.T) {
+		opts := &CreateSchemaOptions{
+			Clone: &Clone{
+				SourceObject: NewAccountObjectIdentifier("sch1"),
+				At: &TimeTravel{
+					Timestamp: Pointer(time.Date(2021, 1, 1, 0, 0, 0, 0, time.UTC)),
+				},
+			},
+		}
+		actual, err := structToSQL(opts)
+		require.NoError(t, err)
+		expected := `CREATE SCHEMA CLONE "sch1" AT (TIMESTAMP => '2021-01-01 00:00:00 +0000 UTC')`
+		assert.Equal(t, expected, actual)
+	})
+
+	t.Run("complete", func(t *testing.T) {
+		opts := &CreateSchemaOptions{
+			OrReplace:                  Bool(true),
```

Comment:
or replace and if exists should not be together,. I know it's unit test, but it documents invalid behaviour.

It's because of a inconsistency we have in tests:
- some of them call validate first and require no errors
- some - like here - go straight to building SQL from struct

Reply:
That's why maybe we should have one assert function for every unit test that would enforce validation and good behaviour ? I'll make it right, but we should consider that

Reply:
Yeah, in my PR I have introduced such function (which goes from validate to building SQL): https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/1968/files#diff-db5a649dfe6533fd95ae49631355518735f67c644d7f9ef52a43b2696b07a8ccR37